packages/ui/src/ui/pages/query-tracker/module/vcs/vcsSlice.ts

```diff
@@ -34,15 +35,24 @@ export type VcsState = {
     listError: boolean;
 };
 
-const getVcsConfig = (): VcsConfig[] => {
+const getVcsData = (): Pick<VcsState, 'vcsConfig' | 'hasVcsSettings'> => {
     const {vcsSettings} = uiSettings;
-    if (!vcsSettings) return [];
 
-    return vcsSettings.map((item) => ({...item, hasToken: false}));
+    if (!vcsSettings) {
+        return {
+            hasVcsSettings: false,
+            vcsConfig: [],
+        };
+    }
+
+    return {
```

Comment:
I don't like that you're introducing a separate `hasVcsSettings` flag here. I'm not sure it's needed here.

I would change the type of the `vcsConfig` variable and explicitly specify undefined if it is not set in the config. So it would be possible to get 3 states (not set, want to hide, set). This way it will be a little more correct and there will not be an extra variable.

Reply:
Sure, will change it

Reply:
Done